http/tus.go

```diff
@@ -0,0 +1,271 @@
+package http
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"sync"
+
+	"github.com/tus/tusd/pkg/filestore"
+	tusd "github.com/tus/tusd/pkg/handler"
+
+	"github.com/filebrowser/filebrowser/v2/settings"
+	"github.com/filebrowser/filebrowser/v2/storage"
+	"github.com/filebrowser/filebrowser/v2/users"
+)
+
+const uploadDirName = ".tmp_upload"
+
+type TusHandler struct {
+	store                *storage.Storage
+	server               *settings.Server
+	settings             *settings.Settings
+	tusdHandlers         map[uint]*tusd.UnroutedHandler
+	notifyNewTusdHandler chan struct{}
+	apiPath              string
+	mutex                *sync.Mutex
+}
+
+func NewTusHandler(store *storage.Storage, server *settings.Server, apiPath string) (TusHandler, error) {
+	tusHandler := TusHandler{}
```

Comment:
Let's use pointer to reduce data coping 
```suggestion
tusHandler := &TusHandler{}
```